### PR TITLE
add a script to remove race data from users incorrectly geolocated to the us

### PR DESCRIPTION
**Pull request by Hamms**
# Description

## Links

- [jira](https://codedotorg.atlassian.net/browse/FND-873)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/oneoff/data_fix/remove_nonus_race_data [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  has_us_teacher = user.student? &&
+    user.teachers.any? do |teacher|
+      teacher.within_united_states? ||
+        teacher.user_school_infos.any? {|usi| usi.school_info.usa?}
+    end
+
+  next if has_us_teacher
+
+  user.races = nil
+  user.save!
```
Comment (bencodeorg) [inline bin/oneoff/data_fix/remove_nonus_race_data:71]: Aren't there a bunch of old users which fail validation for whatever reason? Will that cause a bunch of unnecessary erroring updates? Or is that intentional?

**Review by bencodeorg — APPROVED** [review]
Sweet! Logic looks great to me. Worth running on a clone first to make sure everything goes smoothly?

File: bin/oneoff/data_fix/remove_nonus_race_data (hunk 1 above)
Comment (Hamms) [reply bin/oneoff/data_fix/remove_nonus_race_data:71]: oh, good point. There are, and I'm not sure what we should do about that. @dju90 @joshlory have we figured out a policy that would dictate how we should handle making updates to invalid users?

**Comment by Hamms** [comment]
> Worth running on a clone first to make sure everything goes smoothly?

Good idea!

File: bin/oneoff/data_fix/remove_nonus_race_data (hunk 1 above)
Comment (dju90) [reply bin/oneoff/data_fix/remove_nonus_race_data:71]: Unfortunately, we haven't figured that out yet; there are jira items in our backlog under the "multiauth transition" epic dedicated to figuring out how to deal with these types of accounts. Wrt this work, I'll defer to you and Josh - my only request is that we keep user-felt impact to a minimum.

Comment (Hamms) [reply bin/oneoff/data_fix/remove_nonus_race_data:71]: I created a db clone to do some trials against; i have it right now checking for valid/invalid users and of the 50k users it's checked so far it's only found one invalid one.

We might be able to use this process for valid users, and do something more targeted for invalid users.

Comment (Hamms) [reply bin/oneoff/data_fix/remove_nonus_race_data:71]: Looks like that one invalid user was actually the only one in this whole set! So I'm happy moving forward with this as-is and just manually updating that one user.

**Review by bencodeorg — APPROVED** [review]
Looks good!

**Pull request merged**